nwbwidgets/test/test_plot_func.py

```diff
@@ -0,0 +1,155 @@
+
+
+def test_vis2widget_input_widget():
+        
+    wg = widgets.IntSlider(
+    value=7,
+    min=0,
+    max=10,
+    step=1,
+    description='Test:',
+    disabled=False,
+    continuous_update=False,
+    orientation='horizontal',
+    readout=True,
+    readout_format='d')
+
+    vis2widget(wg)
+
+
+def test_vis2widget_input_figure():
+    
+    data = np.random.rand(160,3)
+    
+    fig=plt.figure(figsize=(10, 5))
+    plt.plot(data)
+    
+    vis2widget(fig)
+
+
+def test_show_subject():
+    
+    data = np.random.rand(160,3)
+    ts = TimeSeries(name='test_timeseries', data=data, unit='m', starting_time=0.0, rate=1.0)
+
+    
+    show_subject(ts)
+
+    
+    
+def test_show_dynamic_table():
+    
+    d = {'col1': [1, 2], 'col2': [3, 4]}
+    DT = DynamicTable.from_dataframe(df=pd.DataFrame(data=d), 
+                                     name='Test Dtable', 
+                                     table_description='no description')
+    
+    show_dynamic_table(DT)
+
+
+
+def test_dict2accordion():
+    
+    data = list(range(100, 200, 10))
+    ts = TimeSeries(name='test_timeseries', data=data, unit='m', starting_time=0.0, rate=1.0)
+    
+    beh_events = BehavioralEvents(time_series=ts)
+    
+    dict2accordion(beh_events.time_series, default_neurodata_vis_spec)
+
+
+def test_show_position():
+    
+    
+    # Create a data interface object (position)
+    spatial_series = SpatialSeries(name = 'position',
+                               data = np.linspace(0, 1, 20),
+                               rate = 50.,
+                               reference_frame = 'starting gate')
+    position = Position(spatial_series = spatial_series)
+    
+    
+    # Test show position function in behavior.py
+    show_position(position, default_neurodata_vis_spec)
+
+
+
+def test_show_behavioral_events():
+    
+    
+    data = list(range(100, 200, 10))
+    ts = TimeSeries(name='test_timeseries', data=data, unit='m', starting_time=0.0, rate=1.0)
+    
+    beh_events = BehavioralEvents(time_series=ts)
+    
+    show_behavioral_events(beh_events, default_neurodata_vis_spec)
+
+    
+def test_show_spatial_series_over_time():
+    
+    
+    # Create a data interface object (position)
+    spatial_series = SpatialSeries(name = 'position',
+                               data = np.linspace(0, 1, 20),
+                               rate = 50.,
+                               reference_frame = 'starting gate')
+    
+    
+    # Test show_spatial_series_over_time function in behavior.py
+    show_spatial_series_over_time(spatial_series)
+
+    
+    
+def test_show_spatial_series():
+    
+    
+    # Create a data interface object (position)
+    spatial_series = SpatialSeries(name = 'position',
+                               data = np.linspace(0, 1, 20),
+                               rate = 50.,
+                               reference_frame = 'starting gate')
+    
+    
+    # Test show_spatial_series function in behavior.py

```

Comment:
You should be able to share code from this function and the one above

Reply:
Do you mean to group them into one class?

Reply:
I think this is fine for now. Let's get this through, and then we can talk about improving the structure with with a `setup` method later